### PR TITLE
e2e: Add a test verifying OLM is working properly by listing its resources

### PR DESCRIPTION
**Pull request by dinhxuanvu**
Add a ginkgo test to list all necessary OLM resources to verify if they are
available and ensure OLM is working properly.

Note: This is an initial verification test for OLM. The test suite will be
expanded further at a later time.

Signed-off-by: Vu Dinh <vdinh@redhat.com>

### CODE REVIEW

File: test/extended/operators/olm.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	yaml "gopkg.in/yaml.v2"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[Feature:Platform][Smoke] OLM should", func() {
```
Comment (bparees) [inline test/extended/operators/olm.go]: Is the Smoke label intentional? I believe that makes this run before all other tests.

Comment (dinhxuanvu) [reply test/extended/operators/olm.go]: I want to follow Clayton's operators test which has Smoke label so they can run together. However, the order in which this test runs doesn't matter. So if you feel the Smoke label is unnecessary, I can remove it.

Comment (dinhxuanvu) [reply test/extended/operators/olm.go]: Removed the Smoke label as it seems unnecessary.

**Pull request merged**